rules/embeds/slideshow.js

```diff
@@ -0,0 +1,114 @@
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+/**
+ * @module rules/embeds/slideshow
+ */
+
+/**
+ * Add support for Slideshow in Markdown, as block syntax.
+ *
+ * The basic syntax is `[slideshow <url1> <url2>]`. E.g., `[slideshow https://assets.digitalocean.com/banners/python.png https://assets.digitalocean.com/banners/javascript.png]`.
+ * Height and width can optionally be set using `[slideshow <url1> <url2> [height] [width]]`. E.g., `[slideshow https://assets.digitalocean.com/banners/python.png https://assets.digitalocean.com/banners/javascript.png 380 560]`.
+ * The default value for height is 270 and for width is 480.
+ *
+ * @example
+ *
+ * [slideshow https://assets.digitalocean.com/banners/python.png https://assets.digitalocean.com/banners/javascript.png]
+ *
+ * <div class="slideshow" style="height: 270px; width: 480px;">
+ *      <div class="action left" onclick="(() => this.parentNode.getElementsByClassName('slides')[0].scrollLeft -= 480)()">&#8249;</div>
+ *      <div class="action right" onclick="(() => this.parentNode.getElementsByClassName('slides')[0].scrollLeft += 480)()">&#8250;</div>
+ *      <div class="slides"><img src="https://assets.digitalocean.com/banners/python.png" alt="Slide #1" /><img src="https://assets.digitalocean.com/banners/javascript.png" alt="Slide #2" /></div>
+ * </div>
+ *
+ * @type {import('markdown-it').PluginSimple}
+ */
+module.exports = md => {
+    /**
+     * Parsing rule for Slideshow markup.
+     *
+     * @type {import('markdown-it/lib/parser_block').RuleBlock}
+     * @private
+     */
+    const slideshowRule = (state, startLine, endLine, silent) => {
+        // If silent, don't replace
+        if (silent) return false;
+
+        // Get current string to consider (just current line)
+        const pos = state.bMarks[startLine] + state.tShift[startLine];
+        const max = state.eMarks[startLine];
+        const currentLine = state.src.substring(pos, max);
+
+        // Perform some non-regex checks for speed
+        if (currentLine.length < 13) return false; // [slideshow a]
+        if (currentLine.slice(0, 11) !== '[slideshow ') return false;
+        if (currentLine[currentLine.length - 1] !== ']') return false;
+
+        // Check for slideshow match
+        const match = currentLine.match(/^\[slideshow((?: \S+)+)?]$/);
+        if (!match) return false;
+
+        // Get the first image
+        const options = match[1];
+        if (!options) return false;
+
+        // Get the integers for dimensions
+        const numbers = options.split(' ').filter(o => o && Number.isInteger(parseInt(o, 10)));
```

Comment:
With this current implementation, this allows for the width/height to be anywhere in the syntax, before/after/mixed with URLs. I'm fine with that, but we should write a set of tests to make sure the behaviour remains going forward, or we should update this logic to specifically expect them at the end.